keps/0017-pipe-tasks.md

```diff
@@ -43,27 +43,36 @@ Allowing to pipe all kind of files (>1Mb) between tasks requires a general-purpo
 
 ## Proposal
 
-This section describes how pipe tasks and files they produce are configured in the operator. Here is a task definition that produces a file that will be stored as a Secret, referenced by `{{.Pipes.Certificate}}` key:
+This section describes how pipe tasks and files they produce are configured in the operator. Here is a resource definition that produces a file that will be stored as a Secret, referenced by `{{.Pipes.Certificate}}` key:
 ```yaml
-steps:
-    - name: bootstrap
-    tasks:
-        - gencert
-            - pipe:
-                - file: /usr/share/MyKey.key
-                  kind: Secret # ConfigMap
-                  key: {{.Pipes.Certificate}}
+tasks:
+  gencert:
+    resources:
```

Comment:
I still feel kind of weird about the fact that pipe is property of resource, it's in my head still kind of a task property when I think about it and now knowing the implementation... You want to define task and that task is piping something to somewhere...

When I was thinking how to approach that I was kind of thinking if we should not rather create a special type of task for pipes, these tasks would have to have one resource (that anyway has to look totally different than any other resource, because it's not full k8s resource) and then the pipe will be property of that Task... Basically Task would either be

```
tasks:
  deploy:
    resources:
      - service.yaml
      - pdb.yaml
      - configmap.yaml
      - metrics-config.yaml
      - health-check.yaml
      - statefulset.yaml
```

or 

```
tasks:
  deploy:
    pipe:
      file: xxx.txt
      kind: Secret
      key: {{Key}}
    resources:
      - gencert.yaml
```

Then we would have a simple validation that if pipe is inside task, only one resource can be present. That way we don't even have to break the API.

Open to hear what you guys think :)

Reply:
Yeah, as you can see, my initial proposal has `pipe` being a property of the _task_ and not the _resource_. However, that would've created this limitation, where a task can only have this one `ContainerSpec` resource (which I'm ok with). A similar issue exists with e.g. output variables which are also per _resource_ and not per _task_. And as @gerred mentioned above, having tasks/resources of different type bring the problem of polymorphic unmarshaling. 

That all being said, I still like your proposal - `pipe` o the task level makes more sense than on the resource level:
```
tasks:
  deploy:
    pipe:
      file: xxx.txt
      kind: Secret
      key: {{Key}}
    resources:
      - gencert.yaml
```
Will adjust the PR 😉

Reply:
I'm ok with this too, it's a lot simpler and the restriction seems ok to me. Need some resources? Use it as another task! That way a task does one thing and one thing only.